src/stream/src/executor/sort.rs

```diff
@@ -184,15 +185,18 @@ impl<S: StateStore> SortExecutor<S> {
                             Op::Insert => {
                                 // For insert operation, we buffer the record in memory.
                                 let row = row_ref.to_owned_row();
-                                let timestamp = row.0.get(self.sort_column_index).ok_or_else(|| {
+                                let timestamp_datum = row.0.get(self.sort_column_index).ok_or_else(|| {
                                     anyhow!(
                                         "column index {} out of range in row {:?}",
                                         self.sort_column_index,
                                         row
                                     )
-                                })?.clone();
-                                let pk = row.by_indices(&self.pk_indices);
-                                self.buffer.insert((timestamp, pk), (row, false));
+                                })?;
+                                // Rows with None timestamp should be ignored.
```

Comment:
Shall we panic here? Because there must be a watermark filter in upstream. Why does the watermark filter pass this row?

Reply:
I just followed what @xx01cyx wrote.

Reply:
> I just followed what @xx01cyx wrote.

I think it's the time to change that. Come on! :-)

Reply:
You are right. It should panic here. Null watermark should not be silently ignored.